driver/xrt/src/simdevice.cpp

```diff
@@ -28,6 +28,11 @@ SimDevice::SimDevice(unsigned int zmqport, unsigned int local_rank) {
   debug("SimDevice connected");
 };
 
+SimDevice::~SimDevice() {
+    if (zmq_ctx.cmd_socket != nullptr)
+       delete zmq_ctx.cmd_socket;
```

Comment:
I think [this](https://github.com/Mellich/ACCL/pull/1) is a better solution than manually deleting the sockets at the end. Can you give that a go?

Reply:
True. Way better C++. Works for me as well. We now can't copy the struct anymore but doing so would have led to other problems with my approach anyways...

Reply:
Yeah don't really think there is a use case for copying the struct, so I think that's fine.